### ISSUE TITLE
Issue with Generics

### ISSUE THREAD

**Issue by mtiller** [danielgtaylor/huma#74]
If I have a model that is generic, then there seems to be an issue resolving the schemas.  For example, if my model is of type `Container[Thing]` and I create my route with:

```go
registry.Get("registry", "Get the registry",
		responses.OK().Model(Container[Thing]{})).Run(func(ctx huma.Context) {

		// The `WriteModel` convenience method handles content negotiation and
		// serializaing the response for you.
		entity := Container[Thing]{}
		entity.Properties = Thing{Name: "Test1"}

		ctx.WriteModel(http.StatusOK, entity)
	})
```

Then my response yields a `Link` header of `</schemas/Container[main.Thing].json>; rel="describedby"`.  By itself, I don't think this is an issue.  But if I try to access the schema that that URL, I get `404 Not Found`.  Presumably this has something to do with the escaping of the characters in the URL or the pattern matching in the router.  Not really sure, but it doesn't find it.

My current solution is to just alias the type in Go, _e.g.,_

```go
type ContainerThing Container[Thing]
```

This seems to resolve the issue.  But I'm not sure I'll have 100% control over such things in nested types.

**Comment by mt35-rs** [comment]
It appears that the issue actually is related to `.` characters in the type name (my example was contrived).  I'm not 100% sure of this, but I'm working on using a regexp in the route for schemas so that the `schema-id` parameter can match examples like `Container[Thing]` (which I think I've succeeded at) as well as `Container[pkg.Thing]` which is causing an issue (I think because the route pattern has a `.` following the route parameter and so it isn't quite getting the right idea about what the prefix is when processing the route tree in `chi`.

In any case, there appears to be another issue (I'll hopefully verify this once I get through this issue).  If I have a type `Container` from a package `pkg` and it is generic and the type parameter is of type `Thing`, the type name constructed by `huma` appears to be `Container[pkg.Thing]` when it should be `pkg.Container[Thing]`.  As I said, I'll have to confirm this and, if true, I'll open a separate issue for this.

**Comment by mt35-rs** [comment]
So, I went down a rabbit hole learning about `chi` routing and I came to the conclusion that the simplest way to deal with this was to simply use a wildcard, `*`, for the schema name and then parse everything in the handler.  Along the way I revamped the error handler as well.  All this is addressed in #81.  I've tested it and it works with all the special characters I've encountered so far.

**Comment by mtiller** [comment]
With #81 merged, I'm closing this.
